rbd: unexport existing KMS structs for vault, AWS and KeyProtect

humblec opened the pull request:
At present the KMS structs are exported and ideally we should be
able to work without exporting the same.

Updates https://github.com/ceph/ceph-csi/issues/2725 

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
/retest ci/centos/upgrade-tests-rbd

nixpanic commented:
This is just a cleanup, there is no requirement in including this in 3.5.0. We should focus on other PRs that add functionality or fix bugs instead. The 3.5.0 release is already a day too late.

humblec commented:
Rebased to devel.. ptal @nixpanic

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.21

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

humblec commented:
@nixpanic @Rakshith-R can you take a look at this ?

Rakshith-R commented on internal/kms/secretskms.go:
```diff
@@ -49,7 +49,7 @@ const (
 )
 
 // SecretsKMS is default KMS implementation that means no KMS is in use.
```
```suggestion
// secretsKMS is default KMS implementation that means no KMS is in use.
```
change the caps in struct comment too, other places too

humblec replied:
@Rakshith-R done

Rakshith-R commented on internal/kms/secretskms.go:
```diff
@@ -69,43 +69,43 @@ func newSecretsKMS(args ProviderInitArgs) (EncryptionKMS, error) {
-// SecretsMetadataKMS is a KMS based on the SecretsKMS, but stores the
+// SecretsMetadataKMS is a KMS based on the secretsKMS, but stores the
 // Data-Encryption-Key (DEK) in the metadata of the volume.
 type SecretsMetadataKMS struct {
```
```suggestion
type secretsMetadataKMS struct {
```

Unexport this struct too ?

humblec replied:
@Rakshith-R done.. ptal,

Rakshith-R reviewed (commented):
unexport `type DEKStoreType string`, `type IntegratedDEK struct{}` and `type DEKStore interface` too ?

humblec commented:
> unexport `type DEKStoreType string`, `type IntegratedDEK struct{}` and `type DEKStore interface` too ?

@Rakshith-R done other both except the interface, which need by outside package?

Rakshith-R commented:
> > unexport `type DEKStoreType string`, `type IntegratedDEK struct{}` and `type DEKStore interface` too ?
> 
> @Rakshith-R done other both except the interface, which need by outside package?

:+1:, there seems to be a stray file `internal/kms/.keyprotect.go.swp` @humblec

humblec commented:
> > > unexport `type DEKStoreType string`, `type IntegratedDEK struct{}` and `type DEKStore interface` too ?
> > 
> > 
> > @Rakshith-R done other both except the interface, which need by outside package?
> 
> +1, there seems to be a stray file `internal/kms/.keyprotect.go.swp` @humblec

ah. multiple window switching .. removed it .. thanks @Rakshith-R

nixpanic commented on internal/kms/kms.go:
```diff
@@ -316,11 +316,11 @@ func GetDefaultKMS(secrets map[string]string) (EncryptionKMS, error) {
 	Destroy()
 
-	// RequiresDEKStore returns the DEKStoreType that is needed to be
+	// RequiresDEKStore returns the dekStoreType that is needed to be
```
This interface is public, so the return type should be public too?

humblec replied:
better to be public

nixpanic commented on internal/kms/kms.go:
```diff
@@ -334,18 +334,18 @@ type EncryptionKMS interface {
 // particular KMS. A KMS might support different DEKStores depending on its
 // configuration.
-type DEKStoreType string
+type dekStoreType string
```
this is part of the public API. I think it is better to export it, just like the consts  below.

humblec replied:
ok.. lets keep it exported. thanks

humblec committed:
rbd: unexport aws kms structs

At present the KMS structs are exported and ideally we should be
able to work without exporting the same.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rbd: unexport KeyProtect kms struct

At present the KMS structs are exported and ideally we should be
able to work without exporting the same.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rbd: Unexport IntegratedDEK struct from kms

This commit unexport IntegratedDEK struct from KMS
implementation

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rbd: Unexport SecretsMetadataKMS struct

This commit unexport SecretsMetadataKMS struct from KMS
implementation

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rbd: Unexport VaultTenantSA struct from KMS implementation

This commit unexport VaultTenantSA struct from KMS implemenation
of Vault KMS.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rbd: unexport vaultTokenSA struct from KMS implementation

This commit unexport the vaultTokenSA from the vault KMS
implementation

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
rbd: unexport SecretsKMS from KMS implementation

This commit unexport SecretsKMS from KMS implementation.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

nixpanic commented:
> At present the KMS structs are exported and ideally we should be able to work without exporting the same.
> 
> Updates #2750

I am wondering if there is a goal that you want to reach with this. The linked #2750 refers to this PR, and that is not helpful. Depending on the goal, `IntegratedKMS` may need to stay exported.

humblec commented:
> > At present the KMS structs are exported and ideally we should be able to work without exporting the same.
> > Updates #2750
> 
> I am wondering if there is a goal that you want to reach with this. The linked #2750 refers to this PR, and that is not helpful. Depending on the goal, `IntegratedKMS` may need to stay exported.

https://github.com/ceph/ceph-csi/issues/2725 is the correct issue link which came out of a review comment   (https://github.com/ceph/ceph-csi/pull/2723#discussion_r772288543). Corrected the link now, may be we will keep `IntegratedDEK` unexported and open it up if we really need? that sound fine to me.

humblec commented:
@nixpanic @Rakshith-R ptal.. thanks .
